src/sql/parts/jobManagement/views/jobsView.component.ts

```diff
@@ -592,40 +594,40 @@ export class JobsViewComponent extends JobManagementView implements OnInit, OnDe
-			await this._jobManagementService.getJobHistory(ownerUri, job.jobId, job.name).then(async(result) => {
-				if (result) {
-					self.jobSteps[job.jobId] = result.steps ? result.steps : [];
-					self.jobAlerts[job.jobId] = result.alerts ? result.alerts : [];
-					self.jobSchedules[job.jobId] = result.schedules ? result.schedules : [];
-					self.jobHistories[job.jobId] = result.histories ? result.histories : [];
-					self._jobCacheObject.setJobSteps(job.jobId, self.jobSteps[job.jobId]);
-					self._jobCacheObject.setJobHistory(job.jobId, self.jobHistories[job.jobId]);
-					self._jobCacheObject.setJobAlerts(job.jobId, self.jobAlerts[job.jobId]);
-					self._jobCacheObject.setJobSchedules(job.jobId, self.jobSchedules[job.jobId]);
-					let jobHistories = self._jobCacheObject.getJobHistory(job.jobId);
-					let previousRuns: sqlops.AgentJobHistoryInfo[];
-					if (jobHistories.length >= 5) {
-						previousRuns = jobHistories.slice(jobHistories.length - 5, jobHistories.length);
-					} else {
-						previousRuns = jobHistories;
-					}
-					// dont create the charts if the tab changed
-					if (!self._didTabChange) {
-						self.createJobChart(job.jobId, previousRuns);
-					}
-					if (self._agentViewComponent.expanded.has(job.jobId)) {
-						let lastJobHistory = jobHistories[jobHistories.length - 1];
-						let item = self.dataView.getItemById(job.jobId + '.error');
-						let noStepsMessage = nls.localize('jobsView.noSteps', 'No Steps available for this job.');
-						let errorMessage = lastJobHistory ? lastJobHistory.message : noStepsMessage;
-						item['name'] = nls.localize('jobsView.error', 'Error: ') + errorMessage;
-						self._agentViewComponent.setExpanded(job.jobId, item['name']);
-						self.dataView.updateItem(job.jobId + '.error', item);
-					}
+		for (let job of jobs) {
+			while (this._didDialogOpen){}
+			let result = await this._jobManagementService.getJobHistory(ownerUri, job.jobId, job.name);
+			if (result) {
+				self.jobSteps[job.jobId] = result.steps ? result.steps : [];
+				self.jobAlerts[job.jobId] = result.alerts ? result.alerts : [];
+				self.jobSchedules[job.jobId] = result.schedules ? result.schedules : [];
+				self.jobHistories[job.jobId] = result.histories ? result.histories : [];
+				self._jobCacheObject.setJobSteps(job.jobId, self.jobSteps[job.jobId]);
+				self._jobCacheObject.setJobHistory(job.jobId, self.jobHistories[job.jobId]);
+				self._jobCacheObject.setJobAlerts(job.jobId, self.jobAlerts[job.jobId]);
+				self._jobCacheObject.setJobSchedules(job.jobId, self.jobSchedules[job.jobId]);
+				let jobHistories = self._jobCacheObject.getJobHistory(job.jobId);
+				let previousRuns: sqlops.AgentJobHistoryInfo[];
+				if (jobHistories.length >= 5) {
+					previousRuns = jobHistories.slice(jobHistories.length - 5, jobHistories.length);
+				} else {
+					previousRuns = jobHistories;
 				}
-			});
-		}));
+				// dont create the charts if the tab changed
+				if (!self._didTabChange) {
```

Comment:
This will introduce a bug, if i switch to a different tab and then back to this tab, the history of the jobs being processed during this time will never be rendered. Similar to my previous comment, you probably don't need this.